shells/pipes-shell/source/verbs/run-arc.js

```diff
@@ -75,10 +69,26 @@ const instantiateRecipe = async (arc, recipe, particleId, particleName) => {
   return true;
 };
 
+const reinstantiateParticle = (arc, particleId, particleName) => {
+  if (particleId) {
+    const particle = arc.activeRecipe.particles.find(p => p.id === particleId);
+    if (particle) {
+      arc.reinstantiateParticle(particle);
+      return true;
+    }
+    warn(`Particle ${particleName} (${particleId} is not found in the active recipe`);    
+  }
+  return false;
+};
+
 const updateParticleInPlan = (plan, particleId, particleName) => {
   if (!!particleId && !!particleName) {
     plan = plan.clone();
-    plan.particles.find(p => p.name === particleName).id = particleId;
+    const particle = plan.particles.find(p => p.name === particleName);
+    if (!particle) {
+      warn(`Cannot find particle ${particleName} in plan = ${plan.toString()}.`);
```

Comment:
Probably should return here or something, otherwise you'll get a null pointer exception below in line 91

Reply:
good catch, thanks, fixed.